backend/typescript/rest/campRoutes.ts

```diff
@@ -82,30 +82,39 @@ campRouter.post(
 );
 
 /* Update a camp */
-campRouter.patch("/:campId", updateCampDtoValidator, async (req, res) => {
-  try {
-    const newCamp = await campService.updateCampById(req.params.campId, {
-      active: req.body.active,
-      ageLower: req.body.ageLower,
-      ageUpper: req.body.ageUpper,
-      campCoordinators: req.body.campCoordinators,
-      campCounsellors: req.body.campCounsellors,
-      name: req.body.name,
-      description: req.body.description,
-      earlyDropoff: req.body.earlyDropoff,
-      latePickup: req.body.latePickup,
-      location: req.body.location,
-      fee: req.body.fee,
-      startTime: req.body.startTime,
-      endTime: req.body.endTime,
-      volunteers: req.body.volunteers,
-    });
-
-    res.status(200).json(newCamp);
-  } catch (error: unknown) {
-    res.status(500).json({ error: getErrorMessage(error) });
-  }
-});
+campRouter.patch(
+  "/:campId",
+  upload.single("file"),
+  updateCampDtoValidator,
+  async (req, res) => {
+    try {
```

Comment:
the camp data should be in req.body.data instead of req.body since we're changing the type of the request body to be form-data instead of json to be able to attach files, take a look at the route for creating camps: https://github.com/uwblueprint/focus-on-nature/blob/a8bf52a03438b19ad5921155939f7f8c6d934ace/backend/typescript/rest/campRoutes.ts#L52 let's also update the validator